packages/react-radio/src/components/RadioGroup/RadioGroup.test.tsx

```diff
@@ -10,14 +10,222 @@ describe('RadioGroup', () => {
+        <Radio value="a" />
+        <Radio value="b" />
+        <Radio value="c" />
+      </RadioGroup>,
+    );
+    rerender(
+      <RadioGroup defaultValue="b">
+        <Radio value="a" />
+        <Radio value="b" />
+        <Radio value="c" />
+      </RadioGroup>,
+    );
+    expect((getByDisplayValue('a') as HTMLInputElement).checked).toBe(false);
+    expect((getByDisplayValue('b') as HTMLInputElement).checked).toBe(false);
+    expect((getByDisplayValue('c') as HTMLInputElement).checked).toBe(true);
+  });
+
+  it('respects value', () => {
+    const { getByDisplayValue } = render(
+      <RadioGroup value="a">
+        <Radio value="a" />
+        <Radio value="b" />
+        <Radio value="c" />
+      </RadioGroup>,
+    );
+    expect((getByDisplayValue('a') as HTMLInputElement).checked).toBe(true);
+    expect((getByDisplayValue('b') as HTMLInputElement).checked).toBe(false);
+    expect((getByDisplayValue('c') as HTMLInputElement).checked).toBe(false);
+  });
+
+  it('respects changes to value', () => {
+    const { rerender, getByDisplayValue } = render(
+      <RadioGroup value="a">
+        <Radio value="a" />
+        <Radio value="b" />
+        <Radio value="c" />
+      </RadioGroup>,
+    );
+    rerender(
+      <RadioGroup value="b">
+        <Radio value="a" />
+        <Radio value="b" />
+        <Radio value="c" />
+      </RadioGroup>,
+    );
+    expect((getByDisplayValue('a') as HTMLInputElement).checked).toBe(false);
+    expect((getByDisplayValue('b') as HTMLInputElement).checked).toBe(true);
+    expect((getByDisplayValue('c') as HTMLInputElement).checked).toBe(false);
+  });
+
+  it('respects Radio defaultChecked', () => {
+    const { getByDisplayValue } = render(
+      <RadioGroup>
+        <Radio value="a" />
+        <Radio value="b" defaultChecked />
+        <Radio value="c" />
+      </RadioGroup>,
+    );
+    expect((getByDisplayValue('a') as HTMLInputElement).checked).toBe(false);
+    expect((getByDisplayValue('b') as HTMLInputElement).checked).toBe(true);
+    expect((getByDisplayValue('c') as HTMLInputElement).checked).toBe(false);
+  });
+
+  it('respects Radio checked', () => {
+    const { getByDisplayValue } = render(
+      <RadioGroup>
+        <Radio value="a" />
+        <Radio value="b" checked />
+        <Radio value="c" />
+      </RadioGroup>,
+    );
+    expect((getByDisplayValue('a') as HTMLInputElement).checked).toBe(false);
+    expect((getByDisplayValue('b') as HTMLInputElement).checked).toBe(true);
+    expect((getByDisplayValue('c') as HTMLInputElement).checked).toBe(false);
+  });
+
+  it('calls onChange with the correct value', () => {
+    const onChange = jest.fn();
+    const { getByDisplayValue } = render(
+      <RadioGroup onChange={onChange}>
+        <Radio value="a" />
+        <Radio value="b" />
+        <Radio value="c" />
+      </RadioGroup>,
+    );
+
+    expect(onChange).toBeCalledTimes(0);
+
+    fireEvent.click(getByDisplayValue('b'));
```

Comment:
Something I learnt recently is that it might be better to use `userEvent` instead of `fireEvent` to more correctly mimic a user's action. Won't block the PR on this but something to take into account for the future.